rescript/get_rdp_data.py

```diff
@@ -0,0 +1,133 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2022-2022, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import tempfile
+import shutil
+import gzip
+# import warnings
+# import hashlib
+# import pandas as pd
+import qiime2
+from urllib.request import urlretrieve
+# from urllib.error import HTTPError
+# from q2_types.feature_data import MixedCaseDNAFASTAFormat, DNAFASTAFormat
+# from q2_feature_table import merge_seqs, merge_taxa
+
+
+def get_rdp_data(ctx,
+                 ref_db='Archaea+Bacteria',
+                 include_species_labels=False,
+                 rank_propagation=True,
+                 ranks=None):
+
+    parse_taxonomy = ctx.get_action('rescript', 'parse_rdp_taxonomy')
+    merge_rdp_seqs = ctx.get_action('feature_table', 'merge_seqs')
+    merge_rdp_taxonomy = ctx.get_action('feature_table', 'merge_taxa')
+
+    # download data from RDP
+    print('\nDownloading and processing raw files may take some time... '
+          'get some coffee.\n')
+    queries = _assemble_rdp_data_urls(ref_db)
+    rdp_seq_data = _retrieve_data_from_rdp(queries)
+
+    num_seq_files = len(rdp_seq_data)
+
+    if num_seq_files == 2:
```

Comment:
it does not seem like there is a need to calculate `num_seq_files`, as the only reason you would have > 1 is if `ref_db=Archea+Bacteria`.

I also do not think that you should need to parse the taxonomies separately and then merge. Couldn't you just concatenate the seq and taxonomy files (if ref_db=Archea+Bacteria), then parse taxonomy once?

If you do it that way then I think most of the code within the conditional statement could be removed and reduced to just a few lines.